engine/src/main/java/org/terasology/engine/GameEngine.java

```diff
@@ -28,6 +28,23 @@
 public interface GameEngine extends AutoCloseable {
 
     /**
+     * @return The current, fine-grained status of the engine.
+     */
+    EngineStatus getStatus();
```

Comment:
State will eventually be gone, don't worry too much about it (will be a little interim confusion). There will only be getStatus() and the few high level states (running, disposed, etc)